encoding/iid/VertexIID.java

```diff
@@ -215,14 +212,15 @@ public String toString() {
         public static final int VALUE_TYPE_LENGTH = 1;
         static final int VALUE_TYPE_INDEX = PrefixIID.LENGTH + VertexIID.Type.LENGTH;
         static final int VALUE_INDEX = VALUE_TYPE_INDEX + VALUE_TYPE_LENGTH;
-        private final Encoding.ValueType valueType;
+        private final Encoding.ValueType<VALUE> valueType;
 
-        private Attribute(ByteArray bytes) {
+        private Attribute(ByteArray bytes, Encoding.ValueType<VALUE> valueType) {
             super(bytes);
-            valueType = Encoding.ValueType.of(bytes.get(PREFIX_W_TYPE_LENGTH));
```

Comment:
we pass in the parametrised `ValueType`, instead of creating a `ValueType` with a `<?>` here